lib/std/os.zig

```diff
@@ -453,11 +454,54 @@ pub fn abort() noreturn {
         windows.kernel32.ExitProcess(3);
     }
     if (!builtin.link_libc and builtin.os.tag == .linux) {
+        // Linux man page wants to first "unblock SIG.ABRT", but this is a footgun
+        // for user-defined signal handlers that want to restore some state in
+        // some program sections and crash in others
+
+        // user installed SIGABRT handler is run, if installed
         raise(SIG.ABRT) catch {};
 
-        // TODO the rest of the implementation of abort() from musl libc here
+        // disable all signal handlers
+        sigprocmask(SIG.BLOCK, &linux.all_mask, null);
+
+        // TODO thread id should be taken from TLS instead of syscall
+        // TODO add thread group to TLS and use tgkill() instead of tkill()
+        // TODO conditionally add global atomic to start code and add CAS below:
+        //var abort_entered = std.atomic.Atomic(bool).init(false);
+        // ensure teardown by one thread
+        //while(abort_entered.compareAndSwap(false, true, .SeqCst, .SeqCst)) {}
```

Comment:
All of the rest of the implementation needs to be ported, or none of it. This lock is needed for the code below.

Reply:
I did port everything needed and used for now the assumption.